lib/netext/httpext/request.go

```diff
@@ -188,6 +198,17 @@ func readResponseBody(
 			resp.Body, respErr = zlib.NewReader(resp.Body)
 		case CompressionTypeGzip:
 			resp.Body, respErr = gzip.NewReader(resp.Body)
+		case CompressionTypeZstd:
+			var zstdecoder *zstd.Decoder
+			zstdecoder, respErr = zstd.NewReader(resp.Body)
+			// zstd.Decoder.Close() doesn't fully implement the io.ReadCloser
+			// interface, so this defer and wrapping is needed
+			defer zstdecoder.Close()
+			resp.Body = ioutil.NopCloser(zstdecoder)
+		case CompressionTypeBr:
+			brotlireader := brotli.NewReader(resp.Body)
+			// brotli.Reader doesn't implement io.ReadCloser, so wrap it
+			resp.Body = ioutil.NopCloser(brotlireader)
```

Comment:
It is also very important that we actually [read the whole body](https://golang.org/pkg/net/http/#Response) which might not happen if there are errors (for example in the decoding). 
So I propose that a `defer` with Reading the whole body and closing it after that is added, and for all decoders to get a NopClosers so that they don't close the body prematurely

Reply:
Thanks for the suggestions. Take a look at 3249d6d6. I think the added `defer` at line 215 handles @MStoykov's comment, though I didn't use `NopCloser` since the decoders need to be closed as well.